src/components/App.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import FolderList from "./FolderList";
+import { getFolders, getUser } from "../api";
+import Header from "./Header";
+import Footer from "./Footer";
+import Profile from "./Profile";
+import SearchBar from "./SearchBar";
+
+function App() {
+  const [items, setItems] = useState([]);
+  const [user, setUser] = useState(null);
+  const [profile, setProfile] = useState(null);
+  const [folderName, setFolderName] = useState(null);
+
+  const handleLoad = async () => {
+    const { folder } = await getFolders();
+    const user = await getUser();
+    setItems(folder.links);
+    setProfile(folder.owner);
+    setFolderName(folder.name);
+    setUser(user);
+  };
+
+  useEffect(() => {
+    handleLoad();
+  }, []);
+
+  return (
+    <div className="App">
+      <Header user={user} />

```

Comment:
커스텀 훅을 만들어 `<Header />`로 props을 없애고 `<header />`안에서 데이터를 페칭하면 좀 더 깔금합니다.

```suggestion
      <Header />
```

```typescript
const useUser = () => {
    const [user, setUser] = useState(null);

    useEffect(() => {
        const fetchUser = async () => {
            const user = await getUser();
            setUser(user);
        };

        fetchUser();
    }, []);

    return user;
};
```